libz-rs-sys/src/gz.rs

```diff
@@ -1651,6 +1649,40 @@ unsafe fn gz_write(state: &mut GzState, mut buf: *const c_void, mut len: usize)
     put
 }
 
+// Compress `len` null bytes to output.
+//
+// # Returns
+//
+// - `Ok` on success.
+// - `Err` on error.
+fn gz_zero(state: &mut GzState, mut len: usize) -> Result<(), ()> {
+    // Consume whatever is left in the input buffer.
+    if state.stream.avail_in != 0 && gz_comp(state, Z_NO_FLUSH).is_err() {
+        return Err(());
+    }
+
+    // Compress `len` zeros.
+    let mut first = true;
+    while len != 0 {
+        let n = cmp::min(state.in_size, len);
+        if first {
+            // TODO safety comment
+            let input_slice = unsafe { slice::from_raw_parts_mut(state.input, n) };
+            input_slice.fill(0u8);
```

Comment:
this is almost certainly unsound. Would https://doc.rust-lang.org/std/primitive.pointer.html#method.write_bytes work?